wallet-new/src/Cardano/Wallet/Server/LegacyPlugins.hs

```diff
@@ -251,3 +240,17 @@ withMiddleware walletConfiguration wrm app = do
         , Throttle.throttlePeriod = fromIntegral $ tsPeriod ts
         , Throttle.throttleBurst = fromIntegral $ tsBurst ts
         }
+
+-- | A @Middleware@ to enable CORS.
+-- When running in debug mode, chances are we want to at least allow CORS to test the API
+-- with a Swagger editor, locally.
+corsMiddleware :: RunMode -> Middleware
+corsMiddleware wrm =
+    if isDebugMode wrm
+        then cors (const $ Just policy)
+        else identity
+  where
+    policy = simpleCorsResourcePolicy
+        { corsRequestHeaders = ["Content-Type"]
+        , corsMethods = "PUT" : simpleMethods
+        }
```

Comment:
I like it a lot. What about even creating a 'Middleware' module when they can live, instead of mixing them up with Plugins :thinking: ?

Reply:
yes - I thought about it. Will move to a seperate module